app/models/user.rb

```diff
@@ -14,4 +14,15 @@ def self.from_twitter(auth)
       user.password = SecureRandom.hex
     end
   end
+
+  def update_with_twitter(auth)
+    self.name = auth.info.nickname if name.nil? || name.blank?
```

Comment:
`#blank?` already checks for nil values.  Check out http://api.rubyonrails.org/classes/Object.html#method-i-blank-3F

You can simplify this to JUST use `#blank?`  Same in `#linked_with_twitter?`